plugin/plaintextmodel.cpp

```diff
@@ -264,6 +276,25 @@ bool PlainTextModel::readLines(
     return false;
 }
 
+QTextCodec *PlainTextModel::detectCodec(const char *encodedCharacters, int length)
```

Comment:
The size of a file returned by `size()` is `qint64` and the length parameter of uchardet_handle_data() is `size_t`. Even if we really don't want to pass gigabytes of data, it's better to respect 64bits length.

Reply:
What do you mean?

Reply:
I mean that if I have a 5GB file named foo.txt, this will crash the code, or have unexpected behaviour because `int length` cannot represent this actual size. That's why `size_t` exists in C and is used to represent file sizes. Qt is not using size_t when calling `size()` for reason I don't know, but they are using an equivalent type which is `qint64`. So in my opinion, you should declare length either `size_t` or `qint64` (maybe the later because the code is mostly using Qt) to be sure not to overflow integers.